src/core/exception.d

```diff
@@ -601,6 +600,22 @@ extern (C) void onUnicodeError( string msg, size_t idx, string file = __FILE__,
     throw new UnicodeException( msg, idx, file, line );
 }
 
+/**
+ * Throws a static instance of $(LREF Error).
+ *
+ * Params:
+ *  msg  = Information about the error.
+ *  file = The name of the file that signaled this error.
+ *  line = The line number on which this error occurred.
+ *
+ * Throws:
+ *  $(LREF Error).
+ */
+void throwStaticError(string msg, string file = __FILE__, size_t line = __LINE__) @trusted pure nothrow @nogc
```

Comment:
This just throws a generic `Error`.  I don't know what the policy is on strongly typed exceptions.

Reply:
What's the reason for this function?

Reply:
See the body of the function `throw staticError!Error(msg, file, line, null);`.  That template instantiates a static `Error` so that it can be used in `@nogc` code.  However, it is `private`.  I created the `throwStaticError` function to keep it private and still allow throwing static `Error`s.

Reply:
To be able to throw an error in `@nogc`.

Reply:
> I created the throwStaticError function to keep the staticError template private and still allow throwing static

What's the advantage of this wrapper function compared to using `staticError` directly? I don't see how introducing a new public symbol to keep an existing symbol private is an advantage. You can make the existing `staticError` `package` to keep it internal to druntime.

Reply:
I can't make it `package` because that would only apply to modules within `core`.

Reply:
Hmm, right. This is in the `object` module which is not inside any package. But again, is this any better than making the existing `staticError` `public`?

Reply:
No, it's not any better IMO.  If we decide to use static `Error`s, I think I'll just follow the existing pattern for other static `Error`s in `core.exception`.  That is, create a strongly typed `Error` and throw it with an `on{whatever}` function.